includes/class-sensei-course.php

```diff
@@ -145,17 +145,81 @@ public function __construct() {
 
 		add_action( 'template_redirect', [ $this, 'setup_single_course_page' ] );
 		add_action( 'sensei_loaded', [ $this, 'add_legacy_course_hooks' ] );
+
+		// Add custom navigation.
+		add_action( 'in_admin_header', [ $this, 'add_custom_navigation' ] );
+		add_action( 'parent_file', [ $this, 'highlight_menu_item' ] );
 	}
 
 	/**
-	 * Register and enqueue scripts that are needed in the backend.
+	 * Highlight the menu item for the course pages.
+	 *
+	 * @param string $file
+	 *
+	 * @return string
+	 */
+	public function highlight_menu_item( $file ) {
+		global $submenu_file;
+		$screen = get_current_screen();
+
+		if ( $screen && in_array( $screen->id, [ 'edit-course', 'edit-course-category', 'course_page_course-order' ], true ) ) {
+			// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+			$submenu_file = 'edit.php?post_type=course';
+		}
+
+		return $file;
+	}
+
+	/**
+	 * Add custom navigation to the admin pages.
+	 *
+	 * @since 1.9.0
+	 */
+	public function add_custom_navigation() {
+		$screen = get_current_screen();
+		if ( ! $screen ) {
+			return;
+		}
+		if ( in_array( $screen->id, [ 'edit-course', 'edit-course-category' ], true ) ) {
+			$this->display_courses_navigation( $screen );
+		}
+	}
+
+	/**
+	 * Display the courses' navigation.
+	 *
+	 * @param WP_Screen $screen
+	 */
+	private function display_courses_navigation( WP_Screen $screen ) {
+		$html  = '<div id="sensei-custom-navigation" class="sensei-custom-navigation">';
+		$html .= '<div class="sensei-custom-navigation__title">';
+		$html .= '<h1>' . __( 'Courses', 'sensei-lms' ) . '</h1>';
+		$html .= '</div>';
+		$html .= '<div class="sensei-custom-navigation__navbar">';
+		$html .= '<a class="sensei-custom-navigation__navbar__navtab ' . ( '' === $screen->taxonomy ? 'active' : '' ) . '" href="' . esc_url( admin_url( 'edit.php?post_type=course' ) ) . '">' . __( 'All Courses', 'sensei-lms' ) . '</a>';
+		$html .= '<a class="sensei-custom-navigation__navbar__navtab ' . ( 'course-category' === $screen->taxonomy ? 'active' : '' ) . '" href="' . esc_url( admin_url( 'edit-tags.php?taxonomy=course-category&post_type=course' ) ) . '">' . __( 'Course Categories', 'sensei-lms' ) . '</a>';
+		$html .= '<a class="page-title-action" href="' . esc_url( admin_url( 'post-new.php?post_type=course' ) ) . '">' . __( 'New course', 'sensei-lms' ) . '</a>';
+		$html .= '<a href="' . esc_url( admin_url( 'edit.php?post_type=course&page=course-order' ) ) . '">' . __( 'Order courses', 'sensei-lms' ) . '</a>';
+		$html .= '<a href="' . esc_url( admin_url( 'edit.php?post_type=course&page=sensei-settings#course-settings' ) ) . '">' . __( 'Course settings', 'sensei-lms' ) . '</a>';
+		$html .= '</div>';
+		$html .= '</div>';
+
+		echo wp_kses( $html, wp_kses_allowed_html( 'post' ) );
```

Comment:
I'm not sure if the `wp_kses` is needed if all the attributes generated by PHP are escaped. Maybe do a normal HTML output (`?><div>...</div><?php`) to make the code a bit more readable?

Reply:
Oh, that's strange, it's not complaining on my end. I've tested it with the following:

```
private function display_courses_navigation( WP_Screen $screen ) {
	?>
	<div id="sensei-custom-navigation" class="sensei-custom-navigation">
		<div class="sensei-custom-navigation__title">
			<h1><?php esc_html_e( 'Courses', 'sensei-lms' ); ?></h1>
		</div>
	</div>
	<?php
}
```

It's not a big thing, so I'm ok with merging this the way it is. :)

Reply:
You're right. It works well. My mistake was that I didn't escape translations properly.
Here is the commit with  my initial "direct output": https://github.com/Automattic/sensei/pull/4672/commits/f832d879e88bb89d734a4ed5306d23a13ba81b55
And here is the result of linting: https://github.com/Automattic/sensei/runs/4913104809?check_suite_focus=true
And fixed version is here: https://github.com/Automattic/sensei/pull/4672/commits/453421bb4a9d42ea5590a770818e17bf668f8fba